The abstract parent class file generated via generator should not be pluralized

abhaynikam opened the pull request:
The change was introduced in #39866

Currently, the file generated via the generator is pluralized
but the parent class is singular.

Example: `bundle exec rails g scaffold Pet name:string --database=animals`

The above command should generate: `apps/models/animals_record.rb`
but the pets model would inherit from `AnimalRecord` as
`"animals".classify` would be `Animal`

This will throw the `uninitialized constant AnimalRecord Did you mean? AnimalsRecord`
error. This PR fixes the issue by making the abstract parent class file as singular.

@eileencodes I found the issue while I was trying something else. Maybe I'll raise a PR for another scenario as well. Thanks 😊

eileencodes reviewed (changes requested):
The abstract class should match the database name regardless of whether it's plural or singular. I'd prefer this PR fix the inherited class than try to singularize everything.

abhaynikam commented:
Got it. I'll fix it right away and send a PR your way. Thanks for the feedback @eileencodes 😊 🎉

abhaynikam committed:
The abstract parent class file generated via generator should not be pluralized

Currently, the file generated via the generator is pluralized
but the parent class is singluar.

example: bundle exec rails g scaffold Pet name:string --database=animals

The above command should generate: apps/models/animals_record.rb

but the pets model would inherit from: `AnimalRecord` as
`"animals".classify` would be `Animal`

This will throw the `uninitialized constant AnimalRecord Did you mean? AnimalsRecord`
error.

abhaynikam commented:
@eileencodes : Done the changes suggested. Could you please review it once?

eileencodes commented:
Looks great, thank you!
